testing/src/main/scala/prometheus4cats/testing/TestingMetricRegistry.scala

```diff
@@ -0,0 +1,426 @@
+/*
+ * Copyright 2022 Permutive
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package prometheus4cats
+package testing
+
+import cats.syntax.all._
+import cats.data.Chain
+import cats.effect.kernel._
+import cats.effect.std.MapRef
+import cats.data.NonEmptySeq
+import prometheus4cats.util.DoubleMetricRegistry
+import scala.concurrent.duration.FiniteDuration
+import TestingMetricRegistry._
+
+class TestingMetricRegistry[F[_]](
+    val underlying: MapRef[F, (String, List[String]), Option[
+      (Int, MetricType, Metric[Double], MapRef[F, List[String], Chain[Double]])
+    ]],
+    val info: MapRef[F, String, Option[(Int, Info[F, Map[Label.Name, String]])]]
```

Comment:
No biggy, but `MapRef`s make me nervous because you can't get all of the contained values. That said I'm not sure how important doing that really is. What do you think?

Reply:
Hmm good question. Related to the discussion below, it is now private so I guess we can keep it for the moment and then if people have a use-case for it then we can always replace it with `Ref[F, Map[...]]`?